### PR TITLE
[monorepo] reproducible contract builds

### PR DESCRIPTION
**Pull request by IIIIllllIIIIllllIIIIllllIIIIllllIIIIll**
https://www.notion.so/elfour/Post-Mortem-Why-did-withdrawals-not-work-0382d5a610f84c17b37e2433f030224e#0cd2367a52c645a28f77ec0901529836

- [ ] upstream the `legacyOutput` patch
- [x] commit `expected-build` and yarn commands for verifying / generating them
- [x] import `expected-build`
- [ ] switch deploy script to use `expected-build`
- [ ] delete truffle

### CODE REVIEW

**Comment by snario** [comment]
Can we commit the minimum amount of information needed for the artifact?

**Comment by IIIIllllIIIIllllIIIIllllIIIIllllIIIIll** [comment]
IMO the minimum is abi and evm.bytecode.object

File: patches/ethereum-waffle+2.0.15.patch [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+index b7b8b6c..fa67905 100644
+--- a/node_modules/ethereum-waffle/dist/compiler/saveOutput.js
++++ b/node_modules/ethereum-waffle/dist/compiler/saveOutput.js
+@@ -111,7 +111,8 @@ function saveOutputCombined(output, config, filesystem) {
+ }
+ exports.saveOutputCombined = saveOutputCombined;
+ function getContent(contractJson, config) {
+-    if (config.legacyOutput || !contractJson.interface) {
++    if (config.legacyOutput) {
++        console.log("adding legacy output");
```
Comment (snario) [inline patches/ethereum-waffle+2.0.15.patch]: Remove console log?

**Comment by cf19drofxots** [comment]
Note that PR https://github.com/counterfactual/monorepo/pull/2276 needs to be re-done (as per the revert) with documentation as described in its comments.

**Comment by kthomas** [comment]
I agree that `abi` and `evm.bytecode`object are minimum viable but it would be nice to have the entire compiler output.

**Pull request closed without merging**